sefaria/helper/schema.py

```diff
@@ -653,35 +654,52 @@ def needs_rewrite(ref, *args):
         except InputError:
             return False
 
-    def rewriter(ref):
+    def rewriter(our_ref):
         """
         Converts each reference from the simple text to what it should be in the complex text based on the mappings.
-        Assumes that both the references in the mappings and the references that are passed into the function
-        are not ranges. For example, for the line old_ref.contains(ref), suppose it is:
-        Ref("Genesis 6-7").contains(Ref("Genesis 6:3-4"))
-        This will evaluate to true but then in the return statement,
-        it will not successfully replace the old_ref_str with new_ref.
-        To deal with ranges in the future, split the ranged refs into starting_ref() and ending_ref()
-        and then use in_terms_of() to get the data necessary to properly translate the range.
-        Ranges that cross from one section to another will be cut to only spanning the first section.
+        :param our_ref: reference in the simple text's form
+        returns reference in the complex text's form
         """
-        if Ref(ref).is_range():
-            print "Currently does not handle ranged refs.  Only handles the starting_ref() of range."
-            ref = Ref(ref).starting_ref()
+        orig_ref = our_ref
+        our_ref = Ref(our_ref)
+        results = []
+        for our_ref in [our_ref.starting_ref(), our_ref.ending_ref()]:
+            for map_ref in mappings:
+                map_ref = Ref(map_ref)
+                print "******"
+                print map_ref
+                print our_ref
+                try:
+                    terms = our_ref.in_terms_of(map_ref)
+                except:
+                    our_ref.in_terms_of(map_ref)
```

Comment:
This looks like a mistake.  
First - the except clause is over general - what exception are you expecting?  
Second, you're catching an exception and then calling the exact same line again - guaranteeing that you'll cause the exception to be raised upward - making the whole try/except redundant.